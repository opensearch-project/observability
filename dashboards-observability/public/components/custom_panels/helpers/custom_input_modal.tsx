/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useState } from 'react';
import {
  EuiButtonEmpty,
  EuiForm,
  EuiModal,
  EuiModalBody,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiOverlayMask,
  EuiFormRow,
  EuiFieldText,
  EuiButton,
} from '@elastic/eui';

/*
 * "CustomInputModalProps" component is used to create a modal with an input filed
 *
 * Props taken in as params are:
 * runModal - function to fetch input field value and trigger closing modal
 * closeModal - function to trigger closing modal
 * titletxt - string as header for title of modal
 * labelTxt - string as header for input field
 * btn1txt - string as content to fill "close button"
 * btn2txt - string as content to fill "confirm button"
 * openPanelName - Default input value for the field
 * helpText - string help for the input field
 * optionalArgs - Arguments needed to pass them to runModal function
 */

type CustomInputModalProps = {
  runModal:
    | ((value: string, value2: string, value3: string, value4: string) => void)
    | ((value: string) => void);
  closeModal: (
    event?: React.KeyboardEvent<HTMLDivElement> | React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => void;
  labelTxt: string;
  titletxt: string;
  btn1txt: string;
  btn2txt: string;
  openPanelName?: string;
  helpText?: string;
  optionalArgs?: string[];
};

export const CustomInputModal = (props: CustomInputModalProps) => {
  const {
    runModal,
    closeModal,
    labelTxt,
    titletxt,
    btn1txt,
    btn2txt,
    openPanelName,
    helpText,
    optionalArgs,
  } = props;
  const [value, setValue] = useState(openPanelName || ''); // sets input value

  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setValue(e.target.value);
  };

  return (
    <EuiOverlayMask>
      <EuiModal onClose={closeModal} initialFocus="[name=input]">
        <EuiModalHeader>
          <EuiModalHeaderTitle>{titletxt}</EuiModalHeaderTitle>
        </EuiModalHeader>

        <EuiModalBody>
          <EuiForm>
            <EuiFormRow label={labelTxt} helpText={helpText}>
              <EuiFieldText
                data-test-subj="customModalFieldText"
                name="input"
                value={value}
                onChange={(e) => onChange(e)}
              />
            </EuiFormRow>
          </EuiForm>
        </EuiModalBody>

        <EuiModalFooter>
          <EuiButtonEmpty onClick={closeModal}>{btn1txt}</EuiButtonEmpty>
          {optionalArgs === undefined ? (
            <EuiButton data-test-subj="runModalButton" onClick={() => runModal(value)} fill>
              {btn2txt}
            </EuiButton>
          ) : (
            <EuiButton
              data-test-subj="runModalButton"
              onClick={() => runModal(value, ...optionalArgs)}
              fill
            >
              {btn2txt}
            </EuiButton>
          )}
        </EuiModalFooter>
      </EuiModal>
    </EuiOverlayMask>
  );
};
