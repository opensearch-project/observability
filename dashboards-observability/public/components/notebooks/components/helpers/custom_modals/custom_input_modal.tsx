/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React, { useState } from 'react';
import {
  EuiButtonEmpty,
  EuiForm,
  EuiModal,
  EuiModalBody,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiOverlayMask,
  EuiFormRow,
  EuiFieldText,
  EuiButton,
} from '@elastic/eui';

/*
 * "CustomInputModalProps" component is used to create a modal with an input filed
 *
 * Props taken in as params are:
 * runModal - function to fetch input field value and trigger closing modal
 * closeModal - function to trigger closing modal
 * titletxt - string as header for title of modal
 * labelTxt - string as header for input field
 * btn1txt - string as content to fill "close button"
 * btn2txt - string as content to fill "confirm button"
 * openNoteName - Default input value for the field
 */
type CustomInputModalProps = {
  runModal: (value: string) => void;
  closeModal: (
    event?: React.KeyboardEvent<HTMLDivElement> | React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => void;
  labelTxt: string;
  titletxt: string;
  btn1txt: string;
  btn2txt: string;
  openNoteName: string;
  helpText: string;
};

export const CustomInputModal = (props: CustomInputModalProps) => {
  const { runModal, closeModal, labelTxt, titletxt, btn1txt, btn2txt, openNoteName, helpText } = props;
  const [value, setValue] = useState(openNoteName || ''); // sets input value

  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setValue(e.target.value);
  };

  return (
    <EuiOverlayMask>
      <EuiModal onClose={closeModal} initialFocus="[name=input]">
        <EuiModalHeader>
          <EuiModalHeaderTitle>{titletxt}</EuiModalHeaderTitle>
        </EuiModalHeader>

        <EuiModalBody>
          <EuiForm>
            <EuiFormRow label={labelTxt} helpText={helpText}>
              <EuiFieldText name="input" value={value} onChange={(e) => onChange(e)} />
            </EuiFormRow>
          </EuiForm>
        </EuiModalBody>

        <EuiModalFooter>
          <EuiButtonEmpty onClick={closeModal}>{btn1txt}</EuiButtonEmpty>
          <EuiButton onClick={() => runModal(value)} fill>
            {btn2txt}
          </EuiButton>
        </EuiModalFooter>
      </EuiModal>
    </EuiOverlayMask>
  );
};
