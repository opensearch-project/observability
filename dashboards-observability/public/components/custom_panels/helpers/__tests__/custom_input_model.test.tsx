/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { CustomInputModal } from '../custom_input_modal';

describe('Custom Input Model component', () => {
  configure({ adapter: new Adapter() });

  it('renders custom input modal with single argument', () => {
    const runModal = jest.fn;
    const closeModal = jest.fn();
    const labelTxt = 'test label';
    const titletxt = 'Input test';
    const btn1txt = 'btn test';
    const btn2txt = 'btn test 2';
    const wrapper = shallow(
      <CustomInputModal
        runModal={runModal}
        closeModal={closeModal}
        labelTxt={labelTxt}
        titletxt={titletxt}
        btn1txt={btn1txt}
        btn2txt={btn2txt}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders custom input modal with multiple arguments', () => {
    const runModal = jest.fn;
    const closeModal = jest.fn();
    const labelTxt = 'test label';
    const titletxt = 'Input test';
    const btn1txt = 'btn test';
    const btn2txt = 'btn test 2';
    const openPanelName = 'Test Panel';
    const helpText = 'Help Text';
    const optionalArgs = ['option1', 'option2', 'option3'];
    const wrapper = shallow(
      <CustomInputModal
        runModal={runModal}
        closeModal={closeModal}
        labelTxt={labelTxt}
        titletxt={titletxt}
        btn1txt={btn1txt}
        btn2txt={btn2txt}
        openPanelName={openPanelName}
        helpText={helpText}
        optionalArgs={optionalArgs}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
