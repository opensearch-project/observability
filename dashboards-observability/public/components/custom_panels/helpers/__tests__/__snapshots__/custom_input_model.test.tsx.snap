// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Input Model component renders custom input modal with multiple arguments 1`] = `
<EuiOverlayMask>
  <EuiModal
    initialFocus="[name=input]"
    onClose={[MockFunction]}
  >
    <EuiModalHeader>
      <EuiModalHeaderTitle>
        Input test
      </EuiModalHeaderTitle>
    </EuiModalHeader>
    <EuiModalBody>
      <EuiForm>
        <EuiFormRow
          describedByIds={Array []}
          display="row"
          fullWidth={false}
          hasChildLabel={true}
          hasEmptyLabelSpace={false}
          helpText="Help Text"
          label="test label"
          labelType="label"
        >
          <EuiFieldText
            data-test-subj="customModalFieldText"
            name="input"
            onChange={[Function]}
            value="Test Panel"
          />
        </EuiFormRow>
      </EuiForm>
    </EuiModalBody>
    <EuiModalFooter>
      <EuiButtonEmpty
        onClick={[MockFunction]}
      >
        btn test
      </EuiButtonEmpty>
      <EuiButton
        data-test-subj="runModalButton"
        fill={true}
        onClick={[Function]}
      >
        btn test 2
      </EuiButton>
    </EuiModalFooter>
  </EuiModal>
</EuiOverlayMask>
`;

exports[`Custom Input Model component renders custom input modal with single argument 1`] = `
<EuiOverlayMask>
  <EuiModal
    initialFocus="[name=input]"
    onClose={[MockFunction]}
  >
    <EuiModalHeader>
      <EuiModalHeaderTitle>
        Input test
      </EuiModalHeaderTitle>
    </EuiModalHeader>
    <EuiModalBody>
      <EuiForm>
        <EuiFormRow
          describedByIds={Array []}
          display="row"
          fullWidth={false}
          hasChildLabel={true}
          hasEmptyLabelSpace={false}
          label="test label"
          labelType="label"
        >
          <EuiFieldText
            data-test-subj="customModalFieldText"
            name="input"
            onChange={[Function]}
            value=""
          />
        </EuiFormRow>
      </EuiForm>
    </EuiModalBody>
    <EuiModalFooter>
      <EuiButtonEmpty
        onClick={[MockFunction]}
      >
        btn test
      </EuiButtonEmpty>
      <EuiButton
        data-test-subj="runModalButton"
        fill={true}
        onClick={[Function]}
      >
        btn test 2
      </EuiButton>
    </EuiModalFooter>
  </EuiModal>
</EuiOverlayMask>
`;
